lib/nerves_hub_ca/certificate_template.ex

```diff
@@ -0,0 +1,85 @@
+defmodule NervesHubCA.CertificateTemplate do
+  import X509.Certificate.Extension
+  alias X509.Certificate.{Template, Validity}
+
+  @user_validity_years 1
+  @device_validity_years 31
+  @serial_number_bytes 20
+
+  @hash :sha256
+  @ec_named_curve :secp256r1
+  @subject_rdn "/O=NervesHub"
+
+  def hash(), do: @hash
+  def ec_named_curve(), do: @ec_named_curve
+
+  def subject_rdn(), do: @subject_rdn
+  @spec user_subject_rdn() :: binary()
+  def user_subject_rdn(), do: Path.join(@subject_rdn, "CN=NervesHub User Certificate")
+  def device_subject_rdn(), do: Path.join(@subject_rdn, "CN=NervesHub Device Certificate")
```

Comment:
We're requiring the CN be the manufacturer's serial number for user created certs. Should we do that here too?

Reply:
I agree, I updated to take the subject from the CSR.